diff --git a/3rdparty/stout/include/stout/os/posix/fork.hpp b/3rdparty/stout/include/stout/os/posix/fork.hpp
index 098224ee4abe73b61f180af3dac5989141e5908a..83caaec603a4b52bde6dcf300b401b3f68d2bfa7 100644
--- a/3rdparty/stout/include/stout/os/posix/fork.hpp
+++ b/3rdparty/stout/include/stout/os/posix/fork.hpp
@@ -342,16 +342,12 @@ private:
       return pid;
     }
 
-    // Set the basic process information.
-    Tree::Memory process;
-    process.pid = getpid();
-    process.parent = getppid();
-    process.group = getpgid(0);
-    process.session = getsid(0);
-    process.set.store(true);
-
-    // Copy it into shared memory.
-    memcpy(tree.memory.get(), &process, sizeof(Tree::Memory));
+    // Set the basic process information into shared memory.
+    tree.memory->pid = getpid();
+    tree.memory->parent = getppid();
+    tree.memory->group = getpgid(0);
+    tree.memory->session = getsid(0);
+    tree.memory->set.store(true);
 
     // Execute the function, if any.
     if (function.isSome()) {
